security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -109,6 +87,88 @@ func testHelper(t *testing.T, testSocket string, cb secretCallback) {
 	t.Fatalf("failed to start grpc server for SDS")
 }
 
+func TestStreamSecretsPush(t *testing.T) {
+	socket := fmt.Sprintf("/tmp/gotest%s.sock", string(uuid.NewUUID()))
+	arg := Options{
+		UDSPath: socket,
+	}
+	st := &mockSecretStore{}
+	server, err := NewServer(arg, st)
+	defer server.Stop()
+
+	if err != nil {
+		t.Fatalf("failed to start grpc server for sds: %v", err)
+	}
+
+	proxyID := "sidecar~127.0.0.1~id2~local"
+	req := &api.DiscoveryRequest{
+		ResourceNames: []string{"test"},
+		Node: &core.Node{
+			Id: proxyID,
+		},
+	}
+	// Try to call the server
+	conn, err := setupConnection(socket)
+	if err != nil {
+		t.Errorf("failed to setup connection to socket %q", socket)
+	}
+	defer conn.Close()
+
+	sdsClient := sds.NewSecretDiscoveryServiceClient(conn)
+	stream, err := sdsClient.StreamSecrets(context.Background())
```

Comment:
Can you also make this `StreamSecrets` a no-op by default and needs explicitly turned on?

We can turn it on in the test separately.